napari/components/_tests/test_add_layers.py

```diff
@@ -105,3 +105,47 @@ def test_add_layers_with_plugins_and_kwargs(layer_data, kwargs):
                 if 'name' not in kwargs:
                     assert layer.name.startswith('mock_path')
             assert layer.source == expected_source
+
+
+def test_add_points_layer_with_different_range_updates_all_slices():
```

Comment:
Thanks for adding more tests to clarify the intended behavior here! My only ask is that we try to think of better names than `point`, `other_point`, `other_point2` - when there were only two layers that was easy enough to understand. But now there are 4 and the test is longer, so it's quite hard to follow what's going on. Alternatively, you could consider splitting this one test into many to capture explicit setups and expected behaviors, but that's a little more work.

Reply:
Changed the names and added  some more comments about the bahavior. I also learned about the one-test=one-assert policy, but often it seems unpractical and leads to a lot of maintenance work and is also not used in many other tests in this project.

Reply:
Explanatory comments are super helpful. In my experience, one behavior (instead of one assert) per test is often easier to maintain than multiple ones (like this test) because you often delete tests when that behavior is no longer needed (which is very clear) instead of modifying a longer one (which is harder to understand for a new reader). But agreed that many tests take the approach you took here and it's fine for this PR.